src/System.Private.Xml.Linq/tests/TreeManipulation/SaveWithFileName.cs

```diff
@@ -0,0 +1,249 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using CoreXml.Test.XLinq;
+using Microsoft.Test.ModuleCore;
+using XmlCoreTest.Common;
+using Xunit;
+
+namespace XLinqTests
+{
+    public class SaveWithFileName
+    {
+        private const string BaseSaveFileName = "baseSave.xml";
+        private const string TestSaveFileName = "testSave";
+        private const string xml = "<PurchaseOrder><Item price=\"100\">Motor<![CDATA[cdata]]><elem>innertext</elem>text<?pi pi pi?></Item></PurchaseOrder>";
+        private const string xmlForXElementWriteContent = "<Item price=\"100\">Motor<![CDATA[cdata]]><elem>innertext</elem>text<?pi pi pi?></Item>";
+
+        //[Variation(Priority = 0, Desc = "Save and validate XDocument")]
+        [Fact]
+        public void writer_1()
+        {
+            XDocument document = new XDocument(new XElement("PurchaseOrder", new XElement("Item", "Motor", new XAttribute("price", "100"),
+                new XCData("cdata"), new XElement("elem", "innertext"), new XText("text"), new XProcessingInstruction("pi", "pi pi"))));
+            SaveBaseline(xml);
+
+            SaveWithFile(document);
+
+            return;
+        }
+
+        //[Variation(Priority = 0, Desc = "Save and validate XElement")]
+        [Fact]
+        public void writer_2()
+        {
+            XElement element = new XElement("PurchaseOrder", new XElement("Item", "Motor", new XAttribute("price", "100"),
+                new XCData("cdata"), new XElement("elem", "innertext"), new XText("text"), new XProcessingInstruction("pi", "pi pi")));
+            SaveBaseline(xml);
+
+            SaveWithFile(element);
+
+            SaveBaseline(xmlForXElementWriteContent);
+
+            return;
+        }
+
+        //[Variation(Priority = 0, Desc = "Save and validate XStreamingElement")]
+        [Fact]
+        public void writer_3()
+        {
+            XStreamingElement element = new XStreamingElement("PurchaseOrder", new XElement("Item", "Motor", new XAttribute("price", "100"),
+                new XCData("cdata"), new XElement("elem", "innertext"), new XText("text"), new XProcessingInstruction("pi", "pi pi")));
+            SaveBaseline(xml);
+
+            SaveWithFile(element);
+
+            SaveBaseline(xmlForXElementWriteContent);
+
+            return;
+        }
+
+        private static string SerializeXDocumentSaveToFile(XNode node) {
+            string filename = System.IO.Path.GetTempFileName();
+            ((XDocument)node).Save(filename);
+            string result;
+            using (System.IO.TextReader tr = new System.IO.StreamReader(filename)) {
+                result = StripOfXmlDeclaration(tr.ReadToEnd());
+            }
+            System.IO.File.Delete(filename);
```

Comment:
When you do, make sure the file was successfully created, otherwise you'll get an exception.